### ISSUE TITLE
Memory leak via repeated file reads on Win32 systems.

### ISSUE THREAD

**Issue by lukebakken** [erlang/otp#5527]
**Describe the bug**

Repeated reads of files into memory cause a leak. This behavior only exists on `win32` systems. In my testing it appears the leak is introduced with Erlang/OTP `23.2`. Version `23.1.5` does not exhibit it.

**To Reproduce**

Please see this repository, which includes instructions:

https://github.com/lukebakken/vesc-1015-memleak

**Expected behavior**

Repeated reads of files via `file:read_file/1` should not leak memory.

**Affected versions**

`23.2` and above.

My system information:

```
Edition	Windows 10 Enterprise
Version	21H1
Installed on	‎10/‎28/‎2021
OS build	19043.1165
Experience	Windows Feature Experience Pack 120.2212.3530.0
```

**Comment by jhogberg** [comment]
Thanks for your report! The problem affects `file:read_file_info/2` as well.

**Comment by lukebakken** [comment]
@jhogberg wow, did you figure out the root cause *already*???

**Comment by jhogberg** [comment]
Yeah, we didn't free an intermediate result in `efile_read_handle_info` on Windows. It was pretty obvious in hindsight. :/

**Comment by jhogberg** [comment]
Fixed in `OTP-23.3.4.10`, thanks again for the report!

**Comment by jhogberg** [comment]
It's waiting on a few other fixes (mainly `ssl` related) that haven't been finished yet. I'll release it early next week if it hasn't already been released by then.

**Comment by lukebakken** [comment]
@jhogberg - FYI, calls to `filelib:is_regular/1` eventually end up calling `file:read_file_info/1` and trigger the leak. While doing a RabbitMQ trace I can see Mnesia calling `filelib:is_regular/1`:

```
(rabbit@bakkenl-z01)4> redbug:start("file:read_file_info/1->return,stack",[{msgs,100},{time,60000}]).
{376,1}
                       
% 09:30:09 <0.1016.0>({mnesia_controller,dump_and_reply,2})
% file:read_file_info("c:/Users/bakkenl/AppData/Roaming/RabbitMQ/db/rabbit@bakkenl-z01-mnesia/PREVIOUS.LOG")
%   filelib:do_is_regular/2 
%   mnesia_log:prepare_log_dump/1 
%   mnesia_dumper:perform_dump/2 
%   mnesia_controller:dump_and_reply/2 
%   unknown function
```

While I can work around some things in RabbitMQ I can't work around that!

**Comment by garazdawi** [comment]
If you are in a rush you can grab the latest windows installer for the maint branch from GitHub actions: https://github.com/erlang/otp/actions/runs/1682189359

**Comment by lukebakken** [comment]
Thanks @garazdawi for pointing that out!

**Comment by jhogberg** [comment]
Sorry about the delay, I've released `24.2.1` now. :)

**Comment by lukebakken** [comment]
No problem! I saw the announcement and have already installed it. I followed up on the mailing list as well -

https://groups.google.com/g/rabbitmq-users/c/UE-wxXerJl8/m/hPdWFlG3AwAJ

Thanks again!
